src/main/resources/assets/computercraft/lua/rom/apis/paintutils.lua

```diff
@@ -9,25 +9,49 @@ for n=1,16 do
     tColourLookup[ string.byte( "0123456789abcdef",n,n ) ] = 2^(n-1)
 end
 
+local function parseLine( tImageArg, sLine )
+    local tLine = {}
+    for x=1,sLine:len() do
+        tLine[x] = tColourLookup[ string.byte(sLine,x,x) ] or 0
+    end
+    table.insert( tImageArg, tLine )
+    return tImageArg
+end
+
+function parseImage( sRawData )
+    if type( sRawData ) ~= "string" then
+        error ( "bad argument #1 (expected string, got " .. type( sRawData ) .. ")" )
+    end
+    local function split( str, delim ) -- modified split function from https://codea.io/talk/discussion/2118/split-a-string-by-return-newline
+        if string.find( str, delim ) == nil then return { str } end
+        local result, pat, lastpos = {}, "(.-)" .. delim .. "()", nil
+        for part, pos in string.gfind( str, pat ) do table.insert( result, part ); lastpos = pos; end
+        table.insert( result, string.sub( str, lastpos ) )
+        return result
+    end
+    local tImage = {}
+    for _, sLine in pairs( split( sPath, "\n" ) ) do -- read each line like original file handling did
+        tImage = parseLine( tImage, sLine )
+    end
+    return tImage
+end
+
 function loadImage( sPath )
     if type( sPath ) ~= "string" then
         error( "bad argument #1 (expected string, got " .. type( sPath ) .. ")", 2 )
     end
 
-    local tImage = {}
     if fs.exists( sPath ) then
-        local file = io.open(sPath, "r" )
+        local tImage = {}
+        local file = io.open( sPath, "r" )
         local sLine = file:read()
+        local sContent = ""
         while sLine do
```

Comment:
In that case, why even *have* parseLine? Or have it return anything, for that matter... remember tImageArg doesn't hold a table, but a pointer leading to one.

loadImage is additionally creating a redundant tImage table that doesn't need to be there.